WordPress/gutenberg
# DateTime: Create TimeInput component and integrate into TimePicker

## What?
* Introduces a simple `TimeInput` component
* Integrates the `TimeInput` into `TimePicker` component

## Why?


Before these changes, there was no option to use the simple time input component.

| Before | After |
|--------|--------|
| <img width="566" alt="Screenshot 2024-04-11 at 12 11 18" src="https://github.com/WordPress/gutenberg/assets/1241413/18ecee01-761b-4845-b098-679e9b4be295"> | <img width="209" alt="Screenshot 2024-04-11 at 12 11 29" src="https://github.com/WordPress/gutenberg/assets/1241413/f15f7bc1-6e9b-408f-be8c-9c98249d512a"> | 


## How?

Here are more details: https://github.com/WordPress/gutenberg/issues/52734


## Testing Instructions

* Go to TimeInput component: `http://localhost:50240/?path=/docs/components-timeinput--docs`
* Play with applying different combination of arguments

--

* Go to TimePicker component: `http://localhost:50240/?path=/docs/components-timepicker--docs`
* Check if everything work on same way as before this changes

Unit tests: Good old unit tests found a bug, which is resolved. ✅ 






### Testing Instructions for Keyboard


## Screenshots or screencast 

![Screen Capture on 2024-04-11 at 12-42-03](https://github.com/WordPress/gutenberg/assets/1241413/ccf90445-bb63-4256-890d-0dcfea352f93)

---

## ✍️ Dev Note

## TimePicker: Time input can be used in isolation

The time input of the `TimePicker` component can now be used in isolation as `<TimePicker.TimeInput />`. In this case, the `value` will be passed as an object in 24-hour format (`{ hours: number, minutes: number }`).
